server.go

```diff
@@ -57,14 +57,24 @@ func socketinit(w http.ResponseWriter, r *http.Request) {
 			log.Printf("Closing: %+v\n", err)
 			return
 		}
-
-		if validateName(message) {
-			c.WriteMessage(mt, []byte("ok"))
-			defer socketHandle(c, message)
-			return
+		
+		switch messageTokens := strings.SplitAfterN(string(message), " ", 2); messageTokens[0] {
+		case "ping":
+			c.WriteMessage(mt, []byte("pong"))
+			continue
+		case "name":
+			if len(messageTokens = 2) && validateName(messageTokens[1]) {
```

Comment:
This is completely wrong. Fix plz

Reply:
Hey how about we cool it with the hostility on github ok?

Reply:
poggie